bin/cqlsh.py

```diff
@@ -286,8 +286,15 @@ def find_zip(libprefix):
     cassandra.connection.ConnectionBusy, cassandra.connection.ProtocolError, cassandra.connection.ConnectionException,
     cassandra.protocol.ErrorMessage, cassandra.protocol.InternalError, cassandra.query.TraceUnavailable
 )
+CQL_TRACED_ERRORS = (
+    cassandra.CoordinationFailure,  # superclass of ReadFailure
+    cassandra.Timeout,              # superclass of ReadTimeout
+    cassandra.OperationTimedOut,
+    cassandra.protocol.InternalError
+)
```

Comment:
It's a little weird to define this as a standalone seq given that it's a proper subset of CQL_ERRORS.  I'm wondering if we aren't better off defining this seq first and then defining CQL_ERRORS as CQL_TRACED_ERRORS + a literal seq containing all the values that aren't in CQL_TRACED_ERRORS:

```python
CQL_TRACED_ERRORS = (
    cassandra.CoordinationFailure,  # superclass of ReadFailure                                                                                                                                                       
    cassandra.Timeout,              # superclass of ReadTimeout                                                                                                                                                       
    cassandra.OperationTimedOut,
    cassandra.protocol.InternalError
)
CQL_ERRORS = CQL_TRACED_ERRORS + (
    cassandra.AlreadyExists, cassandra.AuthenticationFailed,
    cassandra.InvalidRequest, cassandra.Unauthorized,
    cassandra.cluster.NoHostAvailable,
    cassandra.connection.ConnectionBusy, cassandra.connection.ProtocolError, cassandra.connection.ConnectionException,
    cassandra.protocol.ErrorMessage, cassandra.query.TraceUnavailable
)
```

The advantage of that is that you can now include an except block for CQL_TRACED_ERRORS and you can set the value for err there:

```python
    def perform_simple_statement(self, statement):
        if not statement:
            return False, None

        def print_cql_error(err):
            err_msg = ensure_text(err.message if hasattr(err, 'message') else str(err))
            self.printerr(str(err.__class__.__name__) + ": " + err_msg)

        future = self.session.execute_async(statement, trace=self.tracing_enabled)
        result = None
        traced_err = None
        try:
            result = future.result()
        except CQL_TRACED_ERRORS as err:
            traced_err = err
            print_cql_error(err)
        except CQL_ERRORS as err:
            print_cql_error(err)
        except Exception:
            import traceback
            self.printerr(traceback.format_exc())
```

The nice thing about _that_ is that you've already done the isinstance() checks you do below... so your return statement becomes super-simple:

```python
        if result is None:
            if traced_err or trace_all_errors:
                return False, future
            else:
                return False, None
```